proxy/http/HttpSM.cc

```diff
@@ -529,7 +529,7 @@ HttpSM::attach_client_session(ProxyTransaction *client_vc)
   mptcp_state       = netvc->get_mptcp_state();
   client_tcp_reused = !(ua_txn->is_first_transaction());
 
-  if (auto tbs = dynamic_cast<TLSBasicSupport *>(netvc)) {
+  if (auto tbs = static_cast<TLSBasicSupport *>(netvc->get_service(typeid(TLSBasicSupport)))) {
```

Comment:
In any case, this is logically incorrect.  If the type with NetVConnection as a (direct or indirect) base does not also have TLSBasicSupport as a (direct or indirect) base, static_cast will not return null  It will return a garbage, non-null value.